chromium_src/third_party/blink/renderer/core/frame/location.cc

```diff
@@ -0,0 +1,42 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "third_party/blink/renderer/core/frame/location.h"
+
+#include "base/ranges/algorithm.h"
+
+#define ancestorOrigins ancestorOrigins_ChromiumImpl
+
+#include "src/third_party/blink/renderer/core/frame/location.cc"
+
+#undef ancestorOrigins
+
+namespace blink {
+
+DOMStringList* Location::ancestorOrigins() const {
+  DOMStringList* origins = ancestorOrigins_ChromiumImpl();
+
+  auto is_chrome_untrusted = [](const String& origin) {
+    return origin.StartsWith("chrome-untrusted://");
```

Comment:
Once you rebase on top of https://github.com/brave/brave-core/pull/20622, you'll be able to use `origin->Protocol() == "chrome-untrusted"` since we create a `SecurityOrigin` in the for loop.

It's probably safe either way in this case, but in general it's better to avoid doing any comparisons on stringified URLs.